src/libaktualizr/crypto/p11engine.cc

```diff
@@ -145,7 +145,25 @@ boost::filesystem::path P11Engine::findPkcsLibrary() {
 }
 
 PKCS11_SLOT* P11Engine::findTokenSlot() const {
-  PKCS11_SLOT* slot = PKCS11_find_token(ctx_.get(), wslots_.get_slots(), wslots_.get_nslots());
+  unsigned int i, nslot;
+  PKCS11_SLOT* slot = NULL;
+  PKCS11_SLOT* iterslot = NULL;
+  PKCS11_TOKEN* tok;
+  nslot = wslots_.get_nslots();
+
+  if (label_.empty()) {
```

Comment:
I think we need to fallback to the previous search approach `PKCS11_SLOT* slot = PKCS11_find_token(ctx_.get(), wslots_.get_slots(), wslots_.get_nslots())` if the label is empty for backward compatibility.  

Many companies use libaktualizr and obviously they do not have this new config param (`label`) set in sota.toml so returning `nullptr` if the `label` is not specified will break their libaktualizr-based client.

Reply:
I guess you're right, but IMHO returning any initialized token is a bug. It might work if aktualizr is the only thing using pkcs11, but it's an incorrect approach. One mitigation might be attempt to access the token with pin. If this works there is a smaller chance of accessing wrong token.

Reply:
I agree that the current approach is not quite correct, however it became a "feature bug/issue" already :). So we have to make it backward compatible.

Reply:
We can upstream the version with the backward compatibility and then add another commit with prefix [fio extra] that removes it and yields an error if no label.

Reply:
At least I believe all devices registered via lmp-device-register will have a valid 'aktualizr' token label, then I'm not so sure who else might be using without lmp-device-register (but it is indeed good to not break previous users).

Reply:
I just fixed it to be backward compatible - working with the same bug as before.

Reply:
>At least I believe all devices registered via lmp-device-register will have a valid 'aktualizr' token label,

It's about the new label param of the sota config, not the pkcs11 slot label. The currently deployed devices do not have it.